### PR TITLE
Optimize status flow validation

### PR DESCRIPTION
**Pull request by BalaBalaYi**
### What changes were proposed in this pull request?

Add a status flow validation.

### Why are the changes needed?

To avoid corner case like 'running' -> 'pending' which will cause unexpected fault tolerance.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Review by majieyue — APPROVED** [review]
lgtm

**Review by workingloong — APPROVED** [review]
LGTM

File: dlrover/python/master/node/status_flow.py [hunk 1]
```diff
@@ -119,11 +119,39 @@
 ]
 
 
+allowed_transitions = {
```
Comment (workingloong) [inline dlrover/python/master/node/status_flow.py:122]: Constant uses Capital letters, `ALLOWED_TRANSITIONS`.

**Comment by workingloong** [comment]
`To avoid corner case like 'running' -> 'pending' `
Does the pod status change from running to pending ?

**Comment by BalaBalaYi** [comment]
> allowed_transitions

yes. under tkp case.

File: dlrover/python/master/node/status_flow.py (hunk 1 above)
Comment (BalaBalaYi) [reply dlrover/python/master/node/status_flow.py:122]: done

**Review by samplise — APPROVED** [review]
lgtm

**Pull request merged**